src/fsharp/pars.fsy

```diff
@@ -2761,7 +2764,10 @@ classDefnBindings:
 /* The terminator for a 'let ....' binding in the #light syntax */
 hardwhiteDefnBindingsTerminator:
   |  ODECLEND
-     { (fun _ m -> ()) }
+     { let mToken = rhs parseState 1
+       // In LexFilter.fs the IN token could have been transformed to an ODECLEND one.
+       let mIn = if (mToken.EndColumn - mToken.StartColumn) = 2 then Some mToken else None
```

Comment:
@dsyme I just discovered that this little hack doesn't work with:
```fsharp
do
    let e1 = e :?> Collections.DictionaryEntry
    e1.Key, e1.Value
```

Because `e1` has the same length as `in` the check fall apart.
Any other check I can implement to detect that the token was replaced?